lib/Dialect/StandardOps/Ops.cpp

```diff
@@ -2018,6 +2018,61 @@ OpFoldResult SelectOp::fold(ArrayRef<Attribute> operands) {
   return nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// SplatOp
+//===----------------------------------------------------------------------===//
+
+static void print(OpAsmPrinter &p, SplatOp op) {
+  p << "splat " << *op.getOperand();
+  p.printOptionalAttrDict(op.getAttrs());
+  p << " : " << op.getType();
+}
+
+static ParseResult parseSplatOp(OpAsmParser &parser, OperationState &result) {
+  OpAsmParser::OperandType splatValueInfo;
+  ShapedType shapedType;
+
+  return failure(parser.parseOperand(splatValueInfo) ||
+                 parser.parseOptionalAttributeDict(result.attributes) ||
+                 parser.parseColonType(shapedType) ||
+                 parser.resolveOperand(splatValueInfo,
+                                        shapedType.getElementType(),
+                                        result.operands) ||
+                 parser.addTypeToList(shapedType, result.types));
+}
+
+static LogicalResult verify(SplatOp op) {
+  if (!op.getOperand()->getType().isIntOrIndexOrFloat())
+    return op.emitError("operand should be of integer, index, or float type");
+
+  auto type = op.getType();
+
+  if (!type.isa<VectorType>() && !type.isa<TensorType>())
+    return op.emitError("result type should either be a vector or tensor type");
+
+  if (op.getOperand()->getType() != type.cast<ShapedType>().getElementType())
+    return op.emitError("operand should be of elemental type of result type");
+
+  return success();
+}
+
+// Constant folding hook for SplatOp.
+OpFoldResult SplatOp::fold(ArrayRef<Attribute> operands) {
+  assert(operands.size() == 1 && "splat takes one operand");
+
+  auto constOperand = operands.front();
+  if (!constOperand.dyn_cast_or_null<IntegerAttr>() &&
```

Comment:
Change these to isa:

if (!constOperand || !constOperand.isa<> || ...

Reply:
Done.